res/controllers/Traktor-Kontrol-S2-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1336 @@
+  MessageShort.addControl("[EffectRack1_EffectUnit2", "!effectbutton2", 0xD, "B", 0x10, false, this.effectButton);
+  MessageShort.addControl("[EffectRack1_EffectUnit2", "!effectbutton3", 0xD, "B", 0x08, false, this.effectButton);
+
+  MessageShort.addControl("[Channel1]", "!pfl", 0x0C, "B", 0x10, false, this.pflButton);
+  MessageShort.addControl("[EffectRack1_EffectUnit1]", "group_[Channel1]_enable", 0x0E, "B", 0x08);
+  MessageShort.addControl("[EffectRack1_EffectUnit2]", "group_[Channel1]_enable", 0x0E, "B", 0x04);
+
+  MessageShort.addControl("[Channel2]", "!pfl", 0x0A, "B", 0x10, false, this.pflButton);
+  MessageShort.addControl("[EffectRack1_EffectUnit1]", "group_[Channel2]_enable", 0x0E, "B", 0x02);
+  MessageShort.addControl("[EffectRack1_EffectUnit2]", "group_[Channel2]_enable", 0x0E, "B", 0x01);
+  
+  MessageShort.addControl("[Master]", "maximize_library", 0x0F, "B", 0x04, false, this.toggleButton);
+  
+  MessageShort.addControl("[Microphone]", "talkover", 0x0A, "B", 0x08, false, this.toggleButton);
+  
+  engine.makeConnection("[EffectRack1_EffectUnit1]", "show_parameters", TraktorS2MK2.onShowParametersChange);
+  engine.makeConnection("[EffectRack1_EffectUnit2]", "show_parameters", TraktorS2MK2.onShowParametersChange);
+  
+  this.controller.registerInputPacket(MessageShort);
+
+  // Most items in the long message are controls that go from 0-4096.
+  // There are also some 4 bit encoders.
+  MessageLong.addControl("[Channel1]", "rate", 0x07, "H");
+  MessageLong.addControl("[Channel2]", "rate", 0x09, "H");
+  engine.softTakeover("[Channel1]", "rate", true);
+  engine.softTakeover("[Channel2]", "rate", true);
+  MessageLong.addControl("[Channel1]", "!left_encoder", 0x01, "B", 0x0F, false, this.leftEncoder);
+  MessageLong.addControl("[Channel1]", "!right_encoder", 0x01, "B", 0xF0, false, this.rightEncoder);
+  MessageLong.addControl("[Channel2]", "!left_encoder", 0x02, "B", 0xF0, false, this.leftEncoder);
+  MessageLong.addControl("[Channel2]", "!right_encoder", 0x03, "B", 0x0F, false, this.rightEncoder);
+
+  MessageLong.addControl("[EffectRack1_EffectUnit1]", "mix", 0x17, "H");
+  MessageLong.addControl("[EffectRack1_EffectUnit1", "!effectknob1", 0x19, "H", null, false, this.effectKnob);
+  MessageLong.addControl("[EffectRack1_EffectUnit1", "!effectknob2", 0x1B, "H", null, false, this.effectKnob);
+  MessageLong.addControl("[EffectRack1_EffectUnit1", "!effectknob3", 0x1D, "H", null, false, this.effectKnob);
+
+  MessageLong.addControl("[EffectRack1_EffectUnit2]", "mix", 0x1F, "H");
+  MessageLong.addControl("[EffectRack1_EffectUnit2", "!effectknob1", 0x21, "H", null, false, this.effectKnob);
+  MessageLong.addControl("[EffectRack1_EffectUnit2", "!effectknob2", 0x23, "H", null, false, this.effectKnob);
+  MessageLong.addControl("[EffectRack1_EffectUnit2", "!effectknob3", 0x25, "H", null, false, this.effectKnob);
+  
+  for (var i = 1; i < 3; i++) {
```

Comment:
for loops that start at 1, can the end parameter be i <= 2? I find that a little more easy to understand.  (optional)

Reply:
I find that harder to understand personally.